sparkle/src/components/Checkbox.tsx

```diff
@@ -1,122 +1,98 @@
 
 import { CheckIcon, DashIcon } from "@sparkle/icons/solid";
-import { classNames } from "@sparkle/lib/utils";
+import { cn } from "@sparkle/lib/utils";
 
 import { Icon } from "./Icon";
+import { Label } from "./Label";
 
-export type CheckBoxCheckedStatus = "checked" | "unchecked" | "partial";
-
-export interface CheckboxProps {
-  variant?: "selectable" | "checkable";
-  size?: "xs" | "sm";
-  checked?: CheckBoxCheckedStatus;
-  disabled?: boolean;
-  onChange: (checked: boolean) => void;
-  className?: string;
-}
-
-export function Checkbox({
-  variant = "selectable",
-  checked = "unchecked",
-  size = "sm",
-  onChange,
-  className = "",
-  disabled,
-}: CheckboxProps) {
-  let isChecked: boolean;
-  let isPartialChecked: boolean;
-  switch (checked) {
-    case "checked":
-      isChecked = true;
-      isPartialChecked = false;
-      break;
-    case "unchecked":
-      isChecked = false;
-      isPartialChecked = false;
-      break;
-    case "partial":
-      isChecked = true;
-      isPartialChecked = true;
-      break;
-    default:
-      throw new Error("Invalid checked prop");
+const checkboxStyles = cva(
+  cn(
+    "s-shrink-0 s-peer s-border s-border-primary-500 s-text-foreground",
+    "data-[state=checked]:s-text-white data-[state=checked]:s-text-foreground",
+    "focus-visible:s-ring-ring s-ring-offset-background focus-visible:s-outline-none focus-visible:s-ring-2 focus-visible:s-ring-offset-2",
+    "disabled:s-cursor-not-allowed disabled:s-opacity-50"
+  ),
+  {
+    variants: {
+      isPartial: {
+        false: "data-[state=checked]:s-bg-action-500",
+        true: "data-[state=checked]:s-bg-element-700",
+      },
+      size: {
+        xs: "s-h-4 s-w-4 s-rounded",
+        sm: "s-h-5 s-w-5 s-rounded-md",
+      },
+    },
+    defaultVariants: {
+      size: "sm",
+      isPartial: false,
+    },
   }
+);
 
-  const baseStyleClasses = {
-    base: "s-border s-justify-center s-flex s-items-center s-transition-colors s-duration-300 s-ease-out",
-    idle: "s-bg-structure-50 s-border-element-600 s-cursor-pointer",
-    hover:
-      "hover:s-border-action-500 hover:s-bg-action-200 hover:s-text-white/100",
-    disabled: "s-bg-structure-0 s-border-structure-300 s-cursor-default",
-    dark: {
-      base: "dark:s-bg-structure-50-dark dark:s-border-element-500-dark",
-      hover:
-        "dark:hover:s-border-action-500-dark dark:hover:s-bg-action-200-dark",
-      disabled:
-        "dark:s-bg-structure-0-dark dark:s-border-structure-300-dark s-cursor-default",
-    },
-  };
+interface CheckboxProps
+  extends React.ComponentPropsWithoutRef<typeof CheckboxPrimitive.Root>,
+    VariantProps<typeof checkboxStyles> {}
 
-  const baseSizeClasses = {
-    xs: "s-w-4 s-h-4 s-rounded",
-    sm: "s-w-5 s-h-5 s-rounded-md",
-  };
+const Checkbox = React.forwardRef<
+  React.ElementRef<typeof CheckboxPrimitive.Root>,
+  CheckboxProps
+>(({ className, size = "sm", isPartial = false, checked, ...props }, ref) => (
```

Comment:
With CVA you can probably remove this default props.
```suggestion
>(({ className, size, isPartial, checked, ...props }, ref) => (
```